zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupConfig.java

```diff
@@ -92,7 +93,7 @@ private void validate() {
       if (_namespace == null) {
         throw new IllegalArgumentException("Please specify a namespace for this host.");
       }
-      if (_storageProviderClassName.equals(NFS_BACKUP_STORAGE) && _mountPath == null) {
+      if (_storageProviderClassName.equals(NfsBackupStorage.class.getSimpleName()) && _mountPath == null) {
```

Comment:
Actually, let's not modify BackupConfig. The implementation was not 100% complete, so it's implemented in this PR: https://github.com/linkedin/zookeeper/pull/20/files

Also, we should use the fully-qualified name over the simple name. (you can find examples of the fully qualified names in the enum file in the other PR).

Reply:
I reverted back the changes on BackupConfig. After rebasing to your PR, it should work fine.